tests/ExchangeRateProvider/BaseCurrencyProviderTest.php

```diff
@@ -60,4 +61,25 @@ public function providerGetExchangeRate() : array
             ['CAD', 'GBP', '0.727272'],
         ];
     }
+
+    /**
+     * @dataProvider providerExchangeRateType
+     *
+     * @param  BigNumber|float|int|string      $rate
+     */
+    public function testReturnBigNumber($rate)
+    {
+        $configurableProvider = new ConfigurableProvider();
+        $configurableProvider->setExchangeRate('USD', 'EUR', $rate);
+        $baseProvider = new BaseCurrencyProvider($configurableProvider, 'USD');
+
+        $rate = $baseProvider->getExchangeRate('USD', 'EUR');
+
+        $this->assertEquals(BigNumber::of($rate), $rate);
+    }
+
+    public function providerExchangeRateType()
```

Comment:
Can you please name the provider according to the test name? Like `providerReturnBigNumber`